undertow/src/main/java/org/wildfly/extension/undertow/handlers/FileHandler.java

```diff
@@ -110,8 +110,8 @@ public HttpHandler createHandler(final OperationContext context, ModelNode model
         final boolean directoryListing = DIRECTORY_LISTING.resolveModelAttribute(context, model).asBoolean();
         final boolean followSymlink = FOLLOW_SYMLINK.resolveModelAttribute(context, model).asBoolean();
         final boolean caseSensitive = CASE_SENSITIVE.resolveModelAttribute(context, model).asBoolean();
-        final int cacheBufferSize = CACHE_BUFFER_SIZE.resolveModelAttribute(context, model).asInt();
-        final int cacheBuffers = CACHE_BUFFERS.resolveModelAttribute(context, model).asInt();
+        final long cacheBufferSize = (long) CACHE_BUFFER_SIZE.resolveModelAttribute(context, model).asInt();
+        final long cacheBuffers = (long) CACHE_BUFFERS.resolveModelAttribute(context, model).asInt();
```

Comment:
you should use asLong() instead of asInt()

Reply:
@ehsavoie Hello Emmanuel, thanks for your feedback. This one - fixed.